app/models/enterprise_relationship.rb

```diff
@@ -102,6 +109,10 @@ def revoke_all_child_variant_overrides
     child_variant_overrides.update_all(permission_revoked_at: Time.zone.now)
   end
 
+  def destroy_related_exchanges
+    Exchange.where(sender: parent, receiver: child, incoming: true).destroy_all
```

Comment:
I may be ignorant about how this works, but shouldn't we only destroy the exchanges for current/new order cycles, and leave old ones alone?

Reply:
Also, is `ExchangeDeleter` intended to cover this? That way the logic could be de-duplicated in a central place.

Reply:
> shouldn't we only destroy the exchanges for current/new order cycles, and leave old ones alone?

Neal actually did this and I told him not do because people can change the date of order cycles. Or there's the possibility that someone creates an order cycle without dates, then permissions change and then dates are set. The order cycle would then have unpermitted products in there. And I'm not sure if these exchanges would all be duplicated on order cycle clone as well. That's much more likely.

The downside is that we lose the history of what products were in an old order cycle but we got that problem already and don't need to solve it here. I think it's better to honour the permissions correctly and solve the history problem separately, for example keeping a log or just soft-deleting exchanges. And maybe we will not allow to change the dates of passed order cycles at some point.

Reply:
> is ExchangeDeleter intended to cover this? That way the logic could be de-duplicated in a central place.

It was used for this but then the logic became so simple that I decided that it would be better in the model. The ExchangeDeleter is now unused and needs deleting.